src/scripts/main.js

```diff
@@ -1,7 +1,438 @@
+      'field-cell--1024', 'field-cell--2048',
+    );
+
+    switch (cell.textContent) {
+      case '':
+        break;
+      case '2':
+        cell.classList.add('field-cell--2');
+        break;
+      case '4':
+        cell.classList.add('field-cell--4');
+        break;
+      case '8':
+        cell.classList.add('field-cell--8');
+        break;
+      case '16':
+        cell.classList.add('field-cell--16');
+        break;
+      case '32':
+        cell.classList.add('field-cell--32');
+        break;
+      case '64':
+        cell.classList.add('field-cell--64');
+        break;
+      case '128':
+        cell.classList.add('field-cell--128');
+        break;
+      case '256':
+        cell.classList.add('field-cell--256');
+        break;
+      case '512':
+        cell.classList.add('field-cell--512');
+        break;
+      case '1024':
+        cell.classList.add('field-cell--1024');
+        break;
+      case '2048':
+        cell.classList.add('field-cell--2048');
+        messageWin.classList.remove('hidden');
+        break;
+    }
+  });
+};
+
+buttonStart.addEventListener('click', () => {
+  buttonStart.classList.remove('start');
+  buttonStart.textContent = 'Restart';
+  buttonStart.classList.add('restart');
+  messageStart.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  messageLose.classList.add('hidden');
+  score.textContent = '0';
+
+  for (let i = 0; i < cells.length; i++) {
+    cells[i].textContent = '';
+  }
+
+  const randomCellRestart1 = () => {
+    return Math.floor(Math.random() * cells.length);
+  };
+
+  const constRandomCellRestart1 = randomCellRestart1();
+
+  const randomCellRestart2 = () => {
+    let rando;
+
+    do {
+      rando = Math.floor(Math.random() * cells.length);
+    } while (rando === constRandomCellRestart1);
+
+    return rando;
+  };
+
+  const constRandomCellRestart2 = randomCellRestart2();
+
+  cells[constRandomCellRestart1].textContent = (`${twoOrFour()}`);
+  cells[constRandomCellRestart2].textContent = (`${twoOrFour()}`);
+
+  cellStyle();
+});
+
+document.addEventListener('keydown', action => {
+  action.preventDefault();
+
+  let notEmpty;
+  let flyPoints = '0';
+
+  const floatingScore = () => {
+    if (addScore.classList.contains('animationScore')) {
+      addScore.classList.remove('animationScore');
+    }
+
+    void addScore.offsetWidth;
+
+    addScore.classList.remove('hidden');
+    addScore.textContent = `+${flyPoints}`;
+    addScore.classList.add('animationScore');
+
+    addScore.addEventListener('animationend', () => {
+      addScore.classList.add('hidden');
+    });
+  };
+
+  for (const cell of cells) {
+    if (cell.textContent !== '') {
+      notEmpty = true;
+      break;
+    }
+  }
+
+  if (
+    action.key === 'ArrowUp'
+    && notEmpty === true
+  ) {
+    let changesOccurred = false;
+
+    for (let i = 0; i < cells.length; i++) {
+      let target = i;
+
+      const mover = () => {
+        if (
+          target > 3
```

Comment:
quite difficult to understand logic (a lot of unclear conditions in `if`) - try to make it simpler (at least save conditions result in variables with meaningful names)

Also, maybe you can reuse part of the logic inside each arrow button (action.key === 'ArrowDown') handler